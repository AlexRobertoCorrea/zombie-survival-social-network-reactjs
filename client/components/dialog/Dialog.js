import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';

const SimpleDialog = ({ options }) => {
  const {
    open,
    onClose,
    title,
    content,
    leftButtonOnClick,
    leftButtonAction,
    leftButtonLabel,
    disabled,
    rightButtonOnClick,
    rightButtonLabel,
    rightButtonAction
  } = options;
  
  return (
    <Dialog
      open={open}
      onClose={onClose}
    >
      {title && <DialogTitle>{title}</DialogTitle>}
      <DialogContent>
        {content}
      </DialogContent>
  
      <DialogActions>
        {leftButtonLabel &&
          <Button onClick={() => leftButtonOnClick(leftButtonAction)} color="primary">
            {leftButtonLabel}
          </Button>
        }
  
        {rightButtonLabel &&
          <label htmlFor="contained-button-file">
            <Button
              color="primary"
              disabled={disabled}
              onClick={() => rightButtonOnClick(rightButtonAction)}
            >
              {rightButtonLabel}
            </Button>
          </label>
        }
      </DialogActions>
    </Dialog>
  );
};

SimpleDialog.propTypes = {
  options: PropTypes.shape({
    open: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    content: PropTypes.object.isRequired,
    leftButtonOnClick: PropTypes.func,
    leftButtonAction: PropTypes.bool,
    leftButtonLabel: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    rightButtonOnClick: PropTypes.func,
    rightButtonLabel: PropTypes.string.isRequired,
    rightButtonAction: PropTypes.bool
  })
};

SimpleDialog.defaultProps = {
  options: {
    leftButtonOnClick: () => {},
    leftButtonAction: false,
    disabled: false,
    rightButtonOnClick: () => {},
    rightButtonAction: false
  }
};

export default SimpleDialog;
