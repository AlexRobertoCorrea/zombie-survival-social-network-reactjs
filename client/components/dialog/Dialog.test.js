import React from 'react';

import Dialog from './Dialog';

describe('<Dialog />', () => {
  describe('Render', () => {
    describe('When options are filled, a dialog opens', () => {
      const options = {
        open: true,
        onClose: () => {},
        title: 'title',
        content: (<div />),
        leftButtonOnClick: () => {},
        leftButtonAction: false,
        leftButtonLabel: 'test',
        disabled: false,
        rightButtonOnClick: () => {},
        rightButtonLabel: 'test',
        rightButtonAction: true
      };
    
      const dialog = shallow(<Dialog options={options} />);
    
      it('Ensure that dialog works as expected', () => {
        expect(dialog).toMatchSnapshot();
      });
    });
  });
});
