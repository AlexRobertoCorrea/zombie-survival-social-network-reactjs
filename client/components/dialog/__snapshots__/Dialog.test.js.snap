// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> Render When options are filled, a dialog opens Ensure that dialog works as expected 1`] = `
<WithStyles(Dialog)
  onClose={[Function]}
  open={true}
>
  <WithStyles(DialogTitle)>
    title
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <div />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      test
    </WithStyles(Button)>
    <label
      htmlFor="contained-button-file"
    >
      <WithStyles(Button)
        color="primary"
        disabled={false}
        onClick={[Function]}
      >
        test
      </WithStyles(Button)>
    </label>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
